DB/src/main/java/io/deephaven/db/v2/sources/LongAsDateTimeColumnSource.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+ */
+
+package io.deephaven.db.v2.sources;
+
+import io.deephaven.db.tables.utils.DBDateTime;
+import io.deephaven.db.v2.sources.chunk.Attributes.Values;
+import io.deephaven.db.v2.sources.chunk.*;
+import io.deephaven.db.v2.utils.OrderedKeys;
+import io.deephaven.util.QueryConstants;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Reinterpret result {@link ColumnSource} implementations that translates {@link long} to {@code DBDateTime} values.
+ */
+@AbstractColumnSource.IsSerializable(value = true)
+public class LongAsDateTimeColumnSource extends AbstractColumnSource<DBDateTime> implements MutableColumnSourceGetDefaults.ForObject<DBDateTime> {
+
+    private final ColumnSource<Long> alternateColumnSource;
+
+    public LongAsDateTimeColumnSource(ColumnSource<Long> alternateColumnSource) {
+        super(DBDateTime.class);
+        this.alternateColumnSource = alternateColumnSource;
+    }
+
+    @Override
+    public DBDateTime get(long index) {
+        final long longValue = alternateColumnSource.getLong(index);
+        if (longValue == QueryConstants.NULL_LONG) {
+            return null;
+        }
+        return new DBDateTime(longValue);
+    }
+
+    @Override
+    public DBDateTime getPrev(long index) {
+        final long longValue = alternateColumnSource.getPrevLong(index);
+        if (longValue == QueryConstants.NULL_LONG) {
+            return null;
+        }
+        return new DBDateTime(longValue);
+    }
+
+    @Override
+    public boolean isImmutable() {
+        return alternateColumnSource.isImmutable();
+    }
+
+    @Override
+    public <ALTERNATE_DATA_TYPE> boolean allowsReinterpret(@NotNull final Class<ALTERNATE_DATA_TYPE> alternateDataType) {
+        return alternateDataType == long.class || alternateDataType == Long.class;
+    }
+
+    @Override
+    public <ALTERNATE_DATA_TYPE> ColumnSource<ALTERNATE_DATA_TYPE> doReinterpret(@NotNull final Class<ALTERNATE_DATA_TYPE> alternateDataType) throws IllegalArgumentException {
+        //noinspection unchecked
+        return (ColumnSource<ALTERNATE_DATA_TYPE>) alternateColumnSource;
+    }
+
+    private class ToDateTimeFillContext implements FillContext {
+        final FillContext alternateFillContext;
```

Comment:
Same commentary as on the other class; better to fill from get.

Reply:
And same as on the other class, we should fix `io.deephaven.db.v2.sources.DatetimeAsLongColumnSource` too.